matrix/common/request_tracker.py

```diff
@@ -0,0 +1,108 @@
+from enum import Enum
+
+from matrix.common.dynamo_handler import DynamoHandler, DynamoTable, StateTableField, OutputTableField
+
+
+class Subtask(Enum):
+    """
+    Maps request subtasks to corresponding state tracking field in DynamoDB.
+    """
+    DRIVER = StateTableField.COMPLETED_DRIVER_EXECUTIONS
+    MAPPER = StateTableField.COMPLETED_MAPPER_EXECUTIONS
+    WORKER = StateTableField.COMPLETED_WORKER_EXECUTIONS
+    REDUCER = StateTableField.COMPLETED_REDUCER_EXECUTIONS
+    CONVERTER = StateTableField.COMPLETED_CONVERTER_EXECUTIONS
+
+
+class RequestTracker:
```

Comment:
If you want to roll everything into here, I think you'll need a method to increment expected executions. Not needed for handling the one cell SS2 bundles, but maybe file an issue or mark a TODO.

Reply:
Yup, adding now to increment EXPECTED_WORKER... in Mapper